js/data/data_source/data_source.js

```diff
@@ -788,9 +788,15 @@ var DataSource = Class.inherit({
             }
 
             options.take = 1;
-            options.filter = options.filter
-                ? [options.filter, [propName, propValue]]
-                : [propName, propValue];
+
+            if(options.filter) {
+                if(typeUtils.isFunction(options.filter)) {
+                    options.filter = [options.filter];
+                }
```

Comment:
Did you miss `else` here?

Reply:
No, i need to wrap filter function in an array, because DataSource can parse filter [function() { cusom filter }, [filter by index]] as [ getKey(), "=", value ]